src/pip/_internal/commands/install.py

```diff
@@ -660,6 +667,61 @@ def decide_user_install(
     return True
 
 
+def warn_deprecated_install_options(requirement_set, options):
+    # type: (RequirementSet, Optional[List[str]]) -> None
+    """If any location-changing --install-option arguments were passed for
+    requirements or on the command-line, then show a deprecation warning.
+    """
+    def format_options(option_names):
+        # type: (Iterable[str]) -> List[str]
+        return ["--{}".format(name.replace("_", "-")) for name in option_names]
+
+    requirements = (
+        requirement_set.unnamed_requirements +
+        list(requirement_set.requirements.values())
+    )
+
+    offenders = []
+
+    for requirement in requirements:
+        install_options = requirement.options.get("install_options", [])
+        location_options = parse_args(install_options)
```

Comment:
nit: maybe `parse_distutils_args` (via a change of name in `distutils_args.py` or via `from pip._internal.utils.distutils_args import parse_args as parse_distutils_args`) might ease the understanding.

Reply:
Updated [here](https://github.com/pypa/pip/pull/7371/files#diff-29700915cba3bbdab73c73a21f4752d7R30).